Automattic/wp-calypso
# Gutenboarding: add start and complete tracking events for the main flow

## Changes proposed in this Pull Request

See  #41059

- adding methods for start and end
- adding a constant for the flow name (could also be used for API calls, e.g., sites/new)

## Testing instructions

From http://calypso.localhost:3000/new:

Create a site with a new account

Check that `calypso_signup_start` and `calypso_signup_complete` are called with the right params, e.g.,

<img width="877" alt="Screen Shot 2020-04-17 at 11 57 11 am" src="https://user-images.githubusercontent.com/6458278/79524177-2a8b0580-80a3-11ea-9a46-f404d781a035.png"> 

Now create another site, but this time add a `ref` param, e.g., http://calypso.localhost:3000/new?ref=hoochie

Check that `calypso_signup_start` and `calypso_signup_complete` are called with the right params, 
e.g.,

<img width="577" alt="Screen Shot 2020-04-17 at 12 00 16 pm" src="https://user-images.githubusercontent.com/6458278/79524206-40002f80-80a3-11ea-967f-a1b8e16d2d16.png">
